src/sensitivity.jl

```diff
@@ -56,25 +56,27 @@ function plotUnitSensitivity(paramRange, result, realParam, i)
 end
 
 """ This function calculates the simulated cell number for a pair of drugs in their exact concentration.  """
-function calc_cellNum(pDr1, pDr2, g0)
-    tt = 0.0:1.0:96.0
-    g1d1, g2d1, _ = predict(pDr1, g0, tt)
-    g1d2, g2d2, _ = predict(pDr2, g0, tt)
-    normNum = 1.0 .- [(g1d1[end] + g2d1[end]) / g0, (g1d2[end] + g2d2[end]) / g0]
-    combin = -((normNum[1] + normNum[2] - normNum[1] * normNum[2]) .- 1.0) * g0
+function calc_cellNum(effs, dr1_ind, dr2_ind, g0)
+    pDr1 = effs[:, :, dr1_ind]
+    pDr2 = effs[:, :, dr2_ind]
+
+    g1d1, g2d1, _ = predict(pDr1, g0, 96.0)
+    g1d2, g2d2, _ = predict(pDr2, g0, 96.0)
+    normNum = 1.0 .- [(g1d1 + g2d1)/g0, (g1d2 + g2d2)/g0]
+    combin = -((normNum[1] + normNum[2] - normNum[1] * normNum[2]) .- 1.0) * g0 
     return combin
 end
 
 """ Calculates the difference between the bliss_cell number and bliss_params. """
-function calc_diff(Dr1_params, Dr2_params, effs, Dr1Ind, Dr2Ind, conc1_indx, conc2_indx, g0)
-    # Dr1_params/Dr2_params is a 9 x 1 Array, including the 9 parameters of ODE model. 
-    combin = calc_cellNum(Dr1_params, Dr2_params, g0)
+function calc_diff(Hillp, Dr1Ind, Dr2Ind, conc1_indx, conc2_indx, concs, g0)
+    effs = getODEparamsAll(Hillp, concs)
+    combin = calc_cellNum(effs, Dr1Ind, Dr2Ind, g0)
```

Comment:
Why are you passing the indices in just to subset `effs`? This violates the principal of providing a function with only the data it needs.

Reply:
Fixed with the last commit.